libkbfs/interfaces.go

```diff
@@ -1653,12 +1661,29 @@ type MutableBareRootMetadata interface {
 	// BareRootMetadata. This is necessary since newly-created
 	// BareRootMetadata objects don't have enough data to build a
 	// TlfHandle from until the first rekey.
-	FakeInitialRekey(h BareTlfHandle)
+	FakeInitialRekey(c Codec, h BareTlfHandle) (ExtraMetadata, error)
 	// Update initializes the given freshly-created BareRootMetadata object with
 	// the given TlfID and BareTlfHandle. Note that if the given ID/handle are private,
 	// rekeying must be done separately.
 	// Update implements the BareRootMetadata interface for BareRootMetadataV2.
 	Update(tlf TlfID, h BareTlfHandle) error
 	// Returns the TLF key bundles for this metadata at the given key generation.
-	GetTLFKeyBundles(keyGen KeyGen) (*TLFWriterKeyBundle, *TLFReaderKeyBundle, error)
+	// MDv3 TODO: Get rid of this.
+	GetTLFKeyBundles(keyGen KeyGen) (*TLFWriterKeyBundleV2, *TLFReaderKeyBundleV2, error)
+	// GetUserDeviceKeyInfoMaps returns the given user device key info maps for the given
+	// key generation.
+	GetUserDeviceKeyInfoMaps(keyGen KeyGen, extra ExtraMetadata) (
+		readers, writers UserDeviceKeyInfoMap, err error)
+}
+
+// KeyBundleCache is an interface to a key bundle cache for use with v3 metadata.
+type KeyBundleCache interface {
+	// GetTLFReaderKeyBundleV2 returns the TLFReaderKeyBundleV2 for the given TLFReaderKeyBundleID.
+	GetTLFReaderKeyBundleV2(TLFReaderKeyBundleID) (TLFReaderKeyBundleV3, bool)
```

Comment:
The methods are named V2 but they return types of V3. Sounds like they're all supposed to be v3?